core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -92,7 +93,8 @@ public Schema record(Schema record, List<String> names, List<Schema> fields) {
           hasChange = true; // Sub-fields may be different
           filteredFields.add(copyField(field, fieldSchema, fieldId));
         } else {
-          if (isRecord(field.schema())) {
+          if (isRecord(field.schema())
+              && field.schema().getLogicalType() != VariantLogicalType.get()) {
```

Comment:
I think it would be good to have util method similar to `isRecord` (or maybe update `isRecord`?)

This check is somewhat brittle. Why not check `instance VariantLogicalType`?

Reply:
I did that and reverted, thought that would confuse. But I guess it's only used in this class and it would be fine. Let me move to isRecord().